cmake/modules/BuildDPDK.cmake

```diff
@@ -1,3 +1,37 @@
+list(APPEND dpdk_components
```

Comment:
this is not a big deal, but `BuildDPDK` is included using `include()`, so all variables read/written in the scope of the caller of `include(BuildDPDK)`. ideally, i want to avoid polluting the global naming space if possible. and it'd be better to define this variable close to where it's firstly used, i.e. right before the `foreach(c ${dpdk_components})` loop.

so could you move this into `function(do_build_dpdk dpdk_dir)`?

Reply:
Moved both `dpdk_components` and `dpdk_source_dir` definitions closer to where they are used.